connectors/sources/box.py

```diff
@@ -0,0 +1,445 @@
+    name = "Box"
+    service_type = "box"
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+        self.configuration = configuration
+        self.tasks = 0
+        self.is_enterprise = configuration["is_enterprise"]
+        self.queue = MemQueue(maxmemsize=QUEUE_MEM_SIZE, refresh_timeout=120)
+        self.fetchers = ConcurrentTasks(max_concurrency=MAX_CONCURRENCY)
+
+    def _set_internal_logger(self):
+        self.client.set_logger(logger_=self._logger)
+
+    @cached_property
+    def client(self):
+        return BoxClient(configuration=self.configuration)
+
+    def tweak_bulk_options(self, options):
+        """Tweak bulk options as per concurrent downloads support by Box
+
+        Args:
+            options (dict): Config bulker options.
+        """
+        options["concurrent_downloads"] = CONCURRENT_DOWNLOADS
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Box.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "is_enterprise": {
+                "display": "dropdown",
+                "label": "Box Account",
+                "options": [
+                    {"label": "Box Free Account", "value": BOX_FREE},
+                    {"label": "Box Enterprise Account", "value": BOX_ENTERPRISE},
+                ],
+                "order": 1,
+                "type": "str",
+                "value": BOX_FREE,
+            },
+            "client_id": {
+                "label": "Client ID",
+                "order": 2,
+                "type": "str",
+            },
+            "client_secret": {
+                "label": "Client Secret",
+                "order": 3,
+                "sensitive": True,
+                "type": "str",
+            },
+            "refresh_token": {
+                "depends_on": [{"field": "is_enterprise", "value": BOX_FREE}],
+                "label": "Refresh Token",
+                "order": 4,
+                "sensitive": True,
+                "type": "str",
+            },
+            "enterprise_id": {
+                "depends_on": [{"field": "is_enterprise", "value": BOX_ENTERPRISE}],
+                "label": "Enterprise ID",
+                "order": 5,
+                "type": "int",
+            },
+        }
+
+    async def close(self):
+        await self.client.close()
+
+    async def ping(self):
+        try:
+            await self.client.ping()
+            self._logger.info("Successfully connected to Box.")
+        except Exception:
+            self._logger.exception("Error while connecting to Box.")
+            raise
+
+    async def get_users_id(self):
+        async for user in self.client.paginated_call(
+            url=ENDPOINTS["USERS"], params={}, headers={}
+        ):
+            yield user.get("id")
+
+    async def _fetch_files_folders(self, doc_id, user_id=None):
```

Comment:
Does this method sync all folders' files? Or both files and folders?

Reply:
Yes this method sync both files and folders.